### PR TITLE
[develop] Fixes to setup.py

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Following the change in #42103 if Salt is installed via setup.py then the generated _syspaths.py does not contain the HOME_DIR which results in a failure.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
HOME_DIR was not be included in the _syspaths.py generated via setup.py

### New Behavior
HOME_DIR is not included.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -892,6 +894,7 @@ def __init__(self, attrs=None):
         self.salt_spm_formula_dir = None
         self.salt_spm_pillar_dir = None
         self.salt_spm_reactor_dir = None
+        self.salt_home_dir = None
```
Comment (s0undt3ch) [inline setup.py]: You also need a `salt-hime-dir` entry in `SaltDistribution.global_options` or this value is not configurable at install time.

Comment (s0undt3ch) [reply setup.py]: `salt-home-dir`

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: setup.py [hunk 2]
```diff
@@ -866,8 +868,8 @@ class SaltDistribution(distutils.dist.Distribution):
          'Salt\'s pre-configured SPM formulas directory'),
         ('salt-spm-pillar-dir=', None,
          'Salt\'s pre-configured SPM pillar directory'),
-        ('salt-spm-reactor-dir=', None,
-         'Salt\'s pre-configured SPM reactor directory'),
```
Comment (s0undt3ch) [inline setup.py]: Any reason to remove the `salt-spm-reactor-dir`?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (garethgreenaway) [reply setup.py]: Looks like a cut that should have been a copy.

**Review by s0undt3ch — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Fixes https://github.com/saltstack/salt/issues/43562

**Comment by KumarRajnish** [comment]
All,

Due to a misconfiguration on our Office365 tenant, all email addressed to your Ruckus Wireless email address (you@ruckuswireless.com) were delivered to a catch all mailbox.  This incident occurred between 11 PM to 8 AM (PST).  IT is delivering these emails with a “Reply-all” rule to all individuals on the email chain.  If you feel you are missing email attachments, please send email to helpdesk@ruckuswireless.com.

We apologize for any inconvenience this may have caused.

Regards,

IT

**Comment by openfly** [comment]
This appears to impact the 2018.3.0 release

**Pull request merged**